sys/parsoid.js

```diff
@@ -372,10 +372,10 @@ PSP.generateAndSave = function(hyper, req, format, currentContentRes) {
                 body: res.body[format].body
             };
             resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-            var responsePromise = self.wrapContentReq(hyper, req,
-                    P.resolve(resp), format, tid);
             return self.saveParsoidResult(hyper, req, format, tid, res)
             .then(function() {
+                var responsePromise = self.wrapContentReq(hyper, req,
+                        P.resolve(resp), format, tid);
```

Comment:
@gwicke Just it's not 100% clear to me what's happening here and why was this bug exposed only after content-type version bump. Could you point me to a failed test?